File: server/src/main/java/com/ishop/ishoptech/models/company/Company.java
package com.ishop.ishoptech.models.company;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.ishop.ishoptech.models.BaseEntity;
import com.ishop.ishoptech.models.product.Product;
import lombok.*;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import javax.persistence.*;
import java.util.List;

@Data
@Table
@Entity(name = "company")
@AllArgsConstructor
@NoArgsConstructor
@With
@ToString
public class Company extends BaseEntity {

    private String name;
    private String country;
    private String email;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @OneToMany(cascade = CascadeType.REMOVE, mappedBy = "company", fetch = FetchType.LAZY)
    @Fetch(FetchMode.JOIN)
    @ToString.Exclude
    private List<Product> products;

}

File: client/shop/src/app/models/company/Company.ts
export class Company{
  id: number;
  name: string;
  country: string;
  email: string;

  constructor(id: number, name: string, country: string, email: string) {
    this.id = id;
    this.name = name;
    this.country = country;
    this.email = email;
  }
}
